Apps/WebClient/src/Server/Services/DependentService.cs

```diff
@@ -0,0 +1,103 @@
+    using HealthGateway.Common.ErrorHandling;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Delegates;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using HealthGateway.WebClient.Models;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    public class DependentService : IDependentService
+    {
+        private readonly ILogger logger;
+        private readonly IHttpContextAccessor httpContextAccessor;
+        private readonly IPatientDelegate patientDelegate;
+        private readonly IUserDelegateDelegate userDelegateDelegate;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DependentService"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="httpAccessor">The injected http context accessor provider.</param>
+        /// <param name="patientDelegate">The injected patient registry provider.</param>
+        /// <param name="userDelegateDelegate">The User Delegate delegate to interact with the DB.</param>
+        public DependentService(
+            ILogger<DependentService> logger,
+            IHttpContextAccessor httpAccessor,
+            IPatientDelegate patientDelegate,
+            IUserDelegateDelegate userDelegateDelegate)
+        {
+            this.logger = logger;
+            this.httpContextAccessor = httpAccessor;
+            this.patientDelegate = patientDelegate;
+            this.userDelegateDelegate = userDelegateDelegate;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<DependentModel> AddDependent(string delegateHdId, AddDependentRequest addDependentRequest)
+        {
+            this.logger.LogTrace($"Dependent hdid: {delegateHdId}");
+            this.logger.LogDebug("Getting dependent details...");
+            string jwtString = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0];
+            ResourceIdentifier identifier = new ResourceIdentifier("phn", addDependentRequest.PHN);
+            RequestResult<PatientModel> patientResult = this.patientDelegate.GetPatientByIdentifier(identifier, jwtString);
+            if (patientResult.ResourcePayload == null)
+            {
+                return new RequestResult<DependentModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError() { ResultMessage = "Communication Exception when trying to retrieve the Dependent", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Patient) },
+                };
+            }
+
+            // Verify dependent's details entered by user
+            if (!patientResult.ResourcePayload.LastName.Equals(addDependentRequest.LastName, StringComparison.OrdinalIgnoreCase)
```

Comment:
Can you just add a comparison method to the AddDependentRequest to clean this up.

Reply:
@sslaws do you mean that having a method in AddDependentRequest accepting a PatientResourcePayload as the input?

Reply:
updated per our discussion by moving the checks to the AddDependentRequest model